realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProcessor.kt

```diff
@@ -338,4 +344,29 @@ class RealmProcessor : AbstractProcessor() {
 
         return allValid
     }
+
+    // Because library classes are processed separately, there is no guarantee that this method can
+    // see all of the classes necessary to completely validate all of the realm model sets.  If it can find
+    // the fully-qualified class, though, and prove that the generic type is an embedded object,
```

Comment:
Could we maybe reuse https://github.com/realm/realm-java/pull/7220/files#diff-314913e78729941373d8ebb61f525077f7f5cc070da10ea6776bf4dc30c297e9R2416 to derive whether external classes are embedded to always raise this at compile time?